tests/sles4sap/qesapdeployment/deploy.pm

```diff
@@ -7,14 +7,44 @@
 use strict;
 use warnings;
 use Mojo::Base 'publiccloud::basetest';
+use publiccloud::utils qw(get_credentials);
 use testapi;
 use qesapdeployment;
 
+sub wait_for_ssh {
+    my ($host) = @_;
+    my $timeout //= bmwqemu::scale_timeout(600);
+    my $start_time = time();
+    my $check_port = 1;
+
+    # Looping until reaching timeout or passing two conditions :
+    # - SSH port 22 is reachable
+    # - journalctl got message about reaching one of certain targets
+    while ((my $duration = time() - $start_time) < $timeout) {
+        if ($check_port) {
+            $check_port = 0 if (script_run('nc -vz -w 1 ' . $host . ' 22', quiet => 1) == 0);
+        }
+        else {
+            return $duration;
+        }
+        sleep 5;
+    }
+    die 'Timed out while waiting for ssh to be available in the CSP instances';
+}
+
 sub run {
+    if (check_var('PUBLIC_CLOUD_PROVIDER', 'EC2')) {
```

Comment:
can we think about moving lines 36-39 in qesap_prepare_env of lib/qesapdeploy.pm ?

Reply:
Done!

Verification runs:

https://openqaworker15.qa.suse.cz/tests/105697
https://openqaworker15.qa.suse.cz/tests/105698
https://openqaworker15.qa.suse.cz/tests/105699